ext/sodium/libsodium.c

```diff
@@ -558,7 +558,7 @@ PHP_FUNCTION(sodium_add)
 	val = (unsigned char *) Z_STRVAL(*val_zv);
 	val_len = Z_STRLEN(*val_zv);
 	if (val_len != addv_len) {
-		zend_throw_exception(sodium_exception_ce, "values must have the same length", 0);
+		zend_argument_error(sodium_exception_ce, 1, "and argument #2 ($string_2) must have the same size");
```

Comment:
"and argument #2 ($string_2) must be the same long" or just "and argument #2 ($string_2) must have the same length""?

Reply:
`must have the same length` looks like the best option to me.